sdk/storage/azure-storage-file-share/src/samples/java/com/azure/storage/file/share/ShareServiceAsyncJavaDocCodeSamples.java

```diff
@@ -286,4 +288,41 @@ public void generateAccountSas() {
         String sas = fileServiceAsyncClient.generateAccountSas(sasValues);
         // END: com.azure.storage.file.share.ShareServiceAsyncClient.generateAccountSas#AccountSasSignatureValues
     }
+
+    /**
+     * Code snippet for {@link ShareServiceAsyncClient#undeleteShare(String, String)}.
+     */
+    public void undeleteShare() {
+        ShareServiceAsyncClient fileServiceAsyncClient = createAsyncClientWithSASToken();
+        // BEGIN: com.azure.storage.file.share.ShareServiceAsyncClient.undeleteShare#String-String
+        ListSharesOptions listSharesOptions = new ListSharesOptions();
+        listSharesOptions.setIncludeDeleted(true);
+        fileServiceAsyncClient.listShares(listSharesOptions).flatMap(
+            deletedShare -> {
+                Mono<ShareAsyncClient> shareAsyncClient = fileServiceAsyncClient.undeleteShare(
+                    deletedShare.getName(), deletedShare.getVersion());
+                return shareAsyncClient;
+            }
+        ).then().block();
```

Comment:
No need for `.then()`.

```suggestion
        ).block();
```

Reply:
it's a collection (aka Flux) . I can do blockFirst()